### ISSUE TITLE
RCS1176 should be reported only for variables where type can be inferred

### ISSUE THREAD

**Issue by hankovich** [JosefPihrt/Roslynator#1122]
**Product and Version Used**: 4.3.0

The following examples are false positives:
```csharp
// 1
Span<char> charSpan = stackalloc char[size];

// 2
T result = default!;

// 3
var type = typeof(int);
Type? nullableType = type;

// 4
ProblemDetails details = new();
```
